code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -264,7 +264,7 @@
 		return
 
 	var/obj/screen/LL_dir = hud_used.locate_leader
-	if(xeno_caste.caste_flags & CASTE_IS_INTELLIGENT)
+	if(hive.living_xeno_ruler == src || src == X) // No need to track ourselves, especially if we are the hive leader. 
```

Comment:
```suggestion
	if(src == X) // No need to track ourselves.
```
Wouldn't this behave the same?

Reply:
Just future proof, this assume someone else could be a leader that isn't hive ruler.